src/pe/section_table.rs

```diff
@@ -94,6 +131,29 @@ impl SectionTable {
     }
 }
 
+impl ctx::SizeWith<scroll::Endian> for SectionTable {
+    type Units = usize;
+    fn size_with(_ctx: &scroll::Endian) -> usize {
+        SIZEOF_SECTION_TABLE
+    }
+}
+
+impl ctx::IntoCtx<scroll::Endian> for SectionTable {
+    fn into_ctx(self, bytes: &mut [u8], ctx: scroll::Endian) {
+        let offset = &mut 0;
+        bytes.gwrite(&self.name[..], offset).unwrap();
```

Comment:
Normally one implements the TryIntoCtx variant, and just unwraps that implementation here; why isn't that done?

Reply:
Because I'm not familiar enough with scroll. I'll fix it.

Along those lines, why do many of the goblin structures have a `fn parse` instead of or in addition to implementing TryFromCtx?

Reply:
It was for usually api reasons; I wanted the various parseable "things" to have a `pub fn parse`-esque method which made no mention of `scroll` in the signature, so someone wouldn't be forced to import both goblin + scroll. Probably in many cases, e.g., `Elf::parse` it started off _before_ scroll got written :D (I wrote scroll because I was sick of hand writing stupid byte order field populators, essentially, and adding new parsers or structs was tedious and boring, and refactors or bug fixes were extremely annoying - e.g., multiple times there was some typo read the wrong width out into a variable, or etc.).

I called it `parse`, because I like `new` to be infallible construction, so it was just parse. And adding the `try_from_ctx` impl in elf and to a certain extent mach was an afterthought. If they got added (or not) is just no one has had time to go back and them, or felt the need (since most people use the `Object::parse` endpoint afaics.

And that's the whole story about that :laughing: